Python/compile.c

```diff
@@ -7715,19 +7722,21 @@ optimize_and_assemble_code_unit(struct compiler_unit *u, PyObject *const_cache,
     int nparams = (int)PyList_GET_SIZE(u->u_ste->ste_varnames);
     int nlocals = (int)PyDict_GET_SIZE(u->u_metadata.u_varnames);
     assert(u->u_metadata.u_firstlineno);
-    if (_PyCfg_OptimizeCodeUnit(&g, consts, const_cache, code_flags, nlocals,
+    if (_PyCfg_OptimizeCodeUnit(&g, consts, const_cache, nlocals,
                                 nparams, u->u_metadata.u_firstlineno) < 0) {
         goto error;
     }
 
-    /** Assembly **/
-    int nlocalsplus = prepare_localsplus(&u->u_metadata, &g, code_flags);
-    if (nlocalsplus < 0) {
+    int stackdepth = _PyCfg_Stackdepth(&g);
+    if (stackdepth < 0) {
         goto error;
     }
+    assert(stackdepth >= 0);
+    assert(!(IS_GENERATOR(code_flags) && stackdepth == 0));
```

Comment:
Is it possible for this to happen? `RETURN_GENERATOR` and `POP_TOP` have a net stack effect of zero, but we do need at least stack size of 1 to accommodate them. Are we saved here by the IterationError exception handler that's present on every async function?

It may be worth a comment here to clarify a) why we need to ensure this is true, and b) why we can be confident it will always be true.